File: Data-Structures-and-Algorithms/twosum/twosum.py
class Functions(object):
    def twoSum(self, nums, target):
        for i in range(len(nums)):
            index = i
            comp = target - nums[i]
            for j in range(index+1, len(nums)):   
                if nums[j] == comp:
                    return [i, j]
            
                

arr = [1, 3, 5, 9, 3, 12, -5, 6, 12]
k = 13
print(arr)
print(Functions().twoSum(arr,k))


# O(n^2) for time (n x n steps to iterate through)
# O(1) for space (constant extra space)

File: Data-Structures-and-Algorithms/twosum/twosum.js
const twoSum = function(nums, target) {
    let map = {};
    for (var i=0; i < nums.length; i++) {
        let comp = target - nums[i];
        if (map[comp] !== undefined) {
            return [i, map[comp]]
        } else {
            map[nums[i]] = i;
        }
    }
};

arr = [1, 3, 5, 9, 3, 12, -5, 6, 12]
k = 13
console.log(twoSum(arr,k))

// O(N) for time (N steps to iterate through)
// O(1) for space (constant extra space)

